src/main/java/org/fcrepo/auth/webac/LinkHeaderProvider.java

```diff
@@ -69,15 +73,20 @@
         final ListMultimap<String, String> headers = ArrayListMultimap.create();
 
         LOGGER.debug("Adding WebAC Link Header for Resource: {}", resource);
-
-        nodeService.find(internalSession, resource.getPath()).getTriples(translator, PropertiesRdfContext.class)
-        .filter(t -> model.asStatement(t).getPredicate().hasURI(WEBAC_ACCESS_CONTROL_VALUE))
-        .filter(t -> t.getObject().isURI())
-        .forEachRemaining(t -> {
-                    headers.put("Link", Link.fromUri(uriInfo.getBaseUriBuilder()
-                            .path(translator.convert(model.asStatement(t).getObject().asResource()).getPath())
-                            .toString()).rel("acl").build().toString());
-        });
+        // Get the correct Acl for this resource
+        final Optional<Pair<URI, FedoraResource>> acls = WebACRolesProvider.getEffectiveAcl(resource);
+        if (acls.isPresent()) {
+            // If the Acl is present we probably need to use the internal session to get it's URI
+            nodeService.find(internalSession, acls.get().getRight().getPath())
+                    .getTriples(translator, PropertiesRdfContext.class)
+                    .filter(t -> model.asStatement(t).getPredicate().hasURI(WEBAC_ACCESS_CONTROL_VALUE))
```

Comment:
```WEBAC_ACCESS_CONTROL_VALUE``` is a string so I would have to turn it into a Property here. I'm not clear on the implications for the separation of API from implementation, but would this be okay?
```
 nodeService.find(internalSession, a.getRight().getPath())
    .getTriples(translator, PropertiesRdfContext.class)
    .asModel().listObjectsOfProperty(createProperty(WEBAC_ACCESS_CONTROL_VALUE))
    .forEachRemaining(t -> {
        if (t.isURIResource()) {
           final Resource acl = t.asResource();
           final String aclPath = translator.convert(acl).getPath();
           final URI aclUri = uriInfo.getBaseUriBuilder().path(aclPath).build();
           headers.put("Link", Link.fromUri(aclUri).rel("acl").build().toString());
        }
    });
```